Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,149 @@
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+    using System.Data.SqlClient;
+
+    internal class SqlClientDiagnosticSourceListener : IObserver<KeyValuePair<string, object>>, IDisposable
+    {
+        // Event ids defined at:
+        //    https://github.com/dotnet/corefx/blob/master/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs
+
+        public const string DiagnosticListenerName = "SqlClientDiagnosticListener";
+
+        private const string SqlClientPrefix = "System.Data.SqlClient.";
+
+        public const string SqlAfterExecuteCommand = SqlClientPrefix + "WriteCommandAfter";
+        public const string SqlAfterOpenConnection = SqlClientPrefix + "WriteConnectionOpenAfter";
+
+        private readonly TelemetryClient client;
+        private readonly SqlClientDiagnosticSourceSubscriber subscriber;
+
+        private readonly PropertyFetcher connectionFetcher = new PropertyFetcher("Connection");
+        private readonly PropertyFetcher startTimeFetcher = new PropertyFetcher("StartTime");
+        private readonly PropertyFetcher durationFetcher = new PropertyFetcher("Duration");
+
+        public SqlClientDiagnosticSourceListener(TelemetryConfiguration configuration)
+        {
+            this.client = new TelemetryClient(configuration);
+            this.client.Context.GetInternalContext().SdkVersion =
+                SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceCore + ":");
+
+            this.subscriber = new SqlClientDiagnosticSourceSubscriber(this);
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnCompleted()
+        {
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnError(Exception error)
+        {
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnNext(KeyValuePair<string, object> evnt)
+        {
+            switch (evnt.Key)
+            {
+                case SqlAfterOpenConnection:
+                {
+                    this.OnAfterOpenConnection(
+                        (SqlConnection) connectionFetcher.Fetch(evnt.Value),
+                        (DateTimeOffset) this.startTimeFetcher.Fetch(evnt.Value),
+                        (TimeSpan) this.durationFetcher.Fetch(evnt.Value));
+
+                    break;
+                }
+            }
+        }
+
+        public const string SqlConnectionCommandText = "Open";
+
+        private void OnAfterOpenConnection(SqlConnection connection, DateTimeOffset timestamp, TimeSpan duration)
+        {
+            var telemetry = new DependencyTelemetry();
+
+            // TODO: Activities?
```

Comment:
Since it's a 'After' event it probably does not make sense to create Activity here. Activity represents some operation: it's context and duration. And it makes sense if there is something interesting happens between start and stop.
However I probably miss some context on the EF and it's instrumentation and would like to have more info :)